Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/saver/ImageSaver.java

```diff
@@ -134,15 +180,25 @@ private void startTask(ImageSaveTask task) {
         executor.execute(task);
     }
 
-    private void startBundledTaskInternal(String subFolder, List<ImageSaveTask> tasks) {
+    private boolean startBundledTaskInternal(String subFolder, List<ImageSaveTask> tasks) {
+        boolean allSuccess = true;
+
         for (ImageSaveTask task : tasks) {
             PostImage postImage = task.getPostImage();
-            task.setSubFolder(subFolder);
```

Comment:
The setSubFolder call has to be before the deduplicateFile call, otherwise the deduplicateFile call won't have the right location to look through for the deduplication.